### PR TITLE
TESB-30779 - File-backed memory cache repository.

### PR DESCRIPTION
**Pull request by andreasmattes**
The repository provided is a plug-in replacement for the idempotent file repository for use-cases where idempotency is less important but a fast file-backed in-memory repository is needed which deterministically forgets the oldest entries which have not been used or refreshed for the longest time.

### CODE REVIEW

File: camel-core-talend/README.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# This program and the accompanying materials are made available
+# under the terms of the Apache License v2.0
+# which accompanies this distribution, and is available at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+###############################################################################
+Camel Component Project
+====================
+This Project is a template of the Camel component.
+When you create the component project, you need to move the META-INF/services/org/apache/camel/component/helloworld
```
Comment (jgreffe) [inline camel-core-talend/README.txt:15]: Where are located these files?

File: camel-core-talend/pom.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <!--
+        Copyright (C) 2010 Talend Inc. - www.talend.com
+    -->
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.talend.camel</groupId>
+    <artifactId>camel-core-talend</artifactId>
```
Comment (jgreffe) [inline camel-core-talend/pom.xml:11]: What is the purpose of this new module?

File: camel-core-talend/src/main/java/org/talend/camel/processor/idempotent/FileBackedMemoryRepository.java [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+ * then in case of 1st-level cache miss, the underlying file is scanned which may cost additional performance.
+ * So try to find the right balance of the size of the 1st-level cache, the default size is 1000.
+ * The file store has a maximum capacity of 32mb by default (you can turn this off and have unlimited size).
+ * If the file store grows bigger than the maximum capacity, then the {@link #getDropOldestFileStore()} (is default 1000)
+ * number of entries from the file store is dropped to reduce the file store and make room for newer entries.
+ *
+ * @version 
+ */
+@ManagedResource(description = "File backed memory repository")
+public class FileBackedMemoryRepository extends ServiceSupport implements IdempotentRepository<String> {
```
Comment (jgreffe) [inline camel-core-talend/src/main/java/org/talend/camel/processor/idempotent/FileBackedMemoryRepository.java:59]: Which version of camel has been used?
Which changes have been done?
Why not using the apache forks?

**Review by xldai — COMMENTED** [review]
If that's something should be fixed on idempotent file repository of camel-core, the best practice is creating a Camel Jira and maybe check with Colm/Ormar so that the fixes can be reviewed widely and fixed official on Camel side, then we backport the fixes to camel-core-2.24.2.jar for 7.3.1 monthly. also introduce a new "camel-core-talend" seems to me is not the right way.

**Review by jgreffe — APPROVED** [review]

**Pull request merged**